src/main.ts

```diff
@@ -37,22 +42,27 @@ async function run(): Promise<void> {
         break;
       case 'diff':
         const repo = new Repo();
-        const baseFile = await repo.getBaseFile(file);
+        let file1 = await repo.getBaseFile(file);
+        let file2: string | undefined;
 
-        if (baseFile) {
-          cliParams.unshift(baseFile);
+        if (file1) {
+          file2 = file;
+        } else {
+          file1 = file;
         }
 
-        await bump.Diff.run(cliParams.concat(docCliParams)).then(
-          (version: bump.VersionResponse | undefined) => {
-            if (version && isVersionWithDiff(version)) {
-              diff.run(version).catch(handleErrors);
+        await new bump.Diff(config)
+          .run(file1, file2, doc, hub, token)
+          .then((result: bump.VersionResponse | bump.DiffResponse | undefined) => {
+            if (result && isVersionWithDiff(result)) {
```

Comment:
I ask question here but it could be a general question for this PR:

I'm not sure to understand why they are parallel codes with or without Legacy, and in which case which code is run.
`runLegacy` when version is provided, `run` when diff without version ?


Could you enlighten me about this 🙏  💡  ?

Reply:
This has been removed in the latest version of the PR.

The `run` and `runLegacy` was to differentiate the API calls (old API diff to new API diff). However for now we don't have the new API diff, so we stick to the same API calls :).